test/opentelemetry/otlp_processor_unsampled_test.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+# Copyright (c) 2023 SolarWinds, LLC.
+# All rights reserved.
+
+require 'minitest_helper'
+require './lib/solarwinds_apm/opentelemetry'
+require './lib/solarwinds_apm/constants'
+require './lib/solarwinds_apm/support/txn_name_manager'
+require './lib/solarwinds_apm/support/utils'
+require './lib/solarwinds_apm/sampling/sampling_patch'
+require 'opentelemetry-metrics-sdk'
+
+describe 'otlp processor unsampled test' do
```

Comment:
Curious, how is the test ensuring sample vs not sample in this and the other file? Not seeing anything obvious.

Reply:
unsampled test stub the processor on_finish to nil to simulate that there is no span export (as no sample span); I just want to test out if no span sampled, the metrics still have the `sw.transaction` attributes
```
span_processor.stub(:on_finish, nil)
```

Reply:
I see. Since the transaction naming and response time metrics logic is in the OTLPProcessor (not the SmpleSpanProcessor referenced by `span_processor`), it seems stubbing out `span_processor` is irrelevant to testing and would not catch actual issues with transaction name / metrics attribute. Is it possible to instead provide to the OTLPProcessor a span that:

* IsRecording = true but
* Sampled = false

Which would better simulate the case where a request is not traced? This is the premise of our OTLPProcessor--that for every recorded span, regardless sampled or not, we derive the metrics and set correct `sw.transaction` attribute on it.

Reply:
Yes, I have updated to set the flag to 0 (not sampled).